### PR TITLE
Add redirection for HTTP requests sent to HTTPS host/port

### PR DESCRIPTION
**Pull request by es1024**
Add support for accepting HTTP requests and redirecting them to HTTPS on host/port set to accept HTTPS request in order to handle rolling restart cases where some nodes may have HTTPS enabled while others only have HTTP enabled.

Context: yugabyte/yugabyte-db#13779

### CODE REVIEW

File: squeasel.c [hunk 1]
```diff
@@ -3595,7 +3595,8 @@ static int is_valid_port(unsigned int port) {
 // Examples: 80, 443s, 127.0.0.1:3128, 1.2.3.4:8080s
```
Comment (mbautin) [inline squeasel.c]: Perhaps update the `[ip_address:]port[s]` to `[ip_address:]port[s][r]` above and explain what `s` and `r` mean?

File: squeasel.c [hunk 2]
```diff
@@ -3788,6 +3793,12 @@ static int set_uid_option(struct sq_context *ctx) {
 static pthread_mutex_t *ssl_mutexes;
 
 static int sslize(struct sq_connection *conn, SSL_CTX *s, int (*func)(SSL *)) {
+  char first_byte;
+  // TLS handshake always starts with a ContentType byte equal to 22 (RFC8446 5.1).
+  if (!recv(conn->client.sock, &first_byte, 1, MSG_PEEK) || first_byte != 22) {
+    return 0;
+  }
```
Comment (mbautin) [inline squeasel.c]: What exactly are we trying to achieve with this? Is this a bugfix for HTTP connections to an HTTPS port? Can you clarify in the description?

File: squeasel.c [hunk 3]
```diff
@@ -3788,6 +3793,12 @@ static int set_uid_option(struct sq_context *ctx) {
 static pthread_mutex_t *ssl_mutexes;
 
 static int sslize(struct sq_connection *conn, SSL_CTX *s, int (*func)(SSL *)) {
```
Comment (mbautin) [inline squeasel.c:3801]: This function is missing a description, perhaps add it.

**Review by mbautin — CHANGES_REQUESTED** [review]

File: squeasel.c (hunk 2 above)
Comment (es1024) [reply squeasel.c]: Yes, this is so we don't eat up bytes from the socket assuming it's HTTPS before realizing it's HTTP.

**Review by mbautin — APPROVED** [review]

**Pull request merged**